EmpireBenefitsMessage packet

Resaec opened the pull request:
This PR adds de/encode support for the EmpireBenefitsMessage packet.

There is only one test yet to allow for rapid development on top of this packet.
More samples will be tested against to ensure correct behavior.

Resaec committed:
EmpireBenefitsMessage packet

ScrawnyRonnie commented:
I sent that exact packet on login to see what it would do. New pretty messages in chat:
<img width="361" height="123" alt="image" src="https://github.com/user-attachments/assets/e733c7b5-4e14-44d7-9ae3-9494d6c37722" />
And the global map benefits window:
<img width="322" height="305" alt="image" src="https://github.com/user-attachments/assets/776961d9-8a07-4b92-a36a-9a08f7272abd" />

ScrawnyRonnie committed:
benefit builder, apply benefits

Resaec committed:
Merge pull request #1336 from ScrawnyRonnie/empire_benefits_message_packet

Global Domination Has Its Perks

Resaec committed:
Fix variable naming
Added comments
Added enums for ease of use
Typed packet variables
Added 3 more samples to test cases

Resaec committed:
Fix merge
Fix tests

ScrawnyRonnie commented:
All's well after merge and commits
